src/testdir/test_quickfix.vim

```diff
@@ -6718,4 +6718,28 @@ func Test_hardlink_fname()
   call Xtest_hardlink_fname('l')
 endfunc
 
+func Test_event_QufickFixCmdChanged()
+  let g:count = 0
+  autocmd QuickFixChanged * let g:count = g:count + 1
+
+  let g:l = [{'text': 'foo'}] + repeat([{'text': join(range(30))}], 2)
+  copen
+  call feedkeys(":call setqflist(g:l, 'r')\<CR>", "tx")
+  call assert_equal(1, g:count)
+
+  call feedkeys(":call setqflist(g:l, 'r')\<CR>", "tx")
```

Comment:
Instead of using feedkeys(), can you directly use the g:Xsetlist() function?  You can refer to other tests in the test_quickfix.vim file for how both the quickfix and location lists are tested using the X* functions

Reply:
okay will do it later. currently not sure if this is the right direction. Waiting for some feedback. I feel like it doesn't make sense to trigger before the buffer is updated. But the qf buffer can be null, so it may be correct?